blockbin/src/components/InputForm.js

```diff
@@ -1,16 +1,73 @@
 import React, { Component } from 'react';
+import Web3 from 'web3';
 
 import '../App.css';
 
+const CONTRACT_ADDRESS = '0xa022cc6ff8895fd07b03c480d46f20c412716fdc';
+
 class InputForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pastedText: 'What would you like to put on the ~blockchain~?',
+    };
+
+    if (typeof window.web3 !== 'undefined') {
+      console.log('using mist or metamask');
+      const newWeb3 = new Web3(window.web3.currentProvider);
+      this.web3 = newWeb3;
+    } else {
+      console.log('why aren\'t you using mist or metamask?!');
+      // set the provider you want from Web3.providers
+      this.web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
```

Comment:
what is this for? Do we want to support non-metamask users? As a first step, seems fine to just do `window.alert("you are not using metamask, please install it and reload this page")`

Reply:
yeah you're totally right. This only works locally. I've also been thinking about using https://github.com/coopermaruyama/react-web3 - do you think this is a good idea? Wrapping the page in their `Web3Provider` component?

Reply:
I was just saying that right now I wouldn't even bother accommodating for users without metamask. A simple `alert` will do for now. react-web3 would be good to integrate in a separate PR :)